File: JAVA/145. Binary Tree Postorder Traversal.java
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
       List<Integer> l1=new ArrayList();
       return traversal(root,l1);
    }
    public List<Integer>traversal(TreeNode root,List<Integer> arr){
        if(root!=null){           
            traversal(root.left,arr);
            traversal(root.right,arr);
             arr.add(root.val);
        }
        return arr;
    }  
}

File: C++/145. Binary Tree Postorder Traversal.cpp
class Solution {
public:
 vector<int> arr;
void postorder(TreeNode * root)
{
    if(root==NULL)
        return ;
     postorderTraversal(root->left);
        postorderTraversal(root->right);
        arr.push_back(root->val);
}
    vector<int> postorderTraversal(TreeNode* root) {       
        postorder(root);
        return arr;
    }
};
